contracts/src/periphery/EigenDACertVerifierV2.sol

```diff
@@ -0,0 +1,260 @@
+            revert InvalidSecurityThresholds();
+        }
+        securityThresholdsConfirmation = _securityThresholdsV2.confirmationThreshold;
+        securityThresholdsAdversary = _securityThresholdsV2.adversaryThreshold;
+    }
+
+    /**
+     * @notice Verifies a blob cert using the immutable required quorums and security thresholds
+     * @param batchHeader The batch header of the blob
+     * @param blobInclusionInfo The inclusion proof for the blob cert
+     * @param nonSignerStakesAndSignature The nonSignerStakesAndSignature to verify the blob cert against
+     * @param signedQuorumNumbers The signed quorum numbers corresponding to the nonSignerStakesAndSignature
+     */
+    function verifyDACertV2(
+        BatchHeaderV2 calldata batchHeader,
+        BlobInclusionInfo calldata blobInclusionInfo,
+        NonSignerStakesAndSignature calldata nonSignerStakesAndSignature,
+        bytes memory signedQuorumNumbers
+    ) public view {
+        (CertLib.ErrorCode err, bytes memory errParams) = CertLib.verifyDACertV2(
+            _thresholdRegistry(),
+            _signatureVerifier(),
+            _relayRegistry(),
+            batchHeader,
+            blobInclusionInfo,
+            nonSignerStakesAndSignature,
+            _securityThresholds(),
+            _quorumNumbersRequired(),
+            signedQuorumNumbers
+        );
+        CertLib.revertOnError(err, errParams);
+    }
+
+    /**
+     * @notice Verifies a blob cert using a signed batch
+     * @param signedBatch The signed batch to verify the blob cert against
+     * @param blobInclusionInfo The inclusion proof for the blob cert
+     */
+    function verifyDACertV2FromSignedBatch(
+        SignedBatch calldata signedBatch,
+        BlobInclusionInfo calldata blobInclusionInfo
+    ) external view {
+        (CertLib.ErrorCode err, bytes memory errParams) = CertLib.verifyDACertV2FromSignedBatch(
+            _thresholdRegistry(),
+            _signatureVerifier(),
+            _relayRegistry(),
+            _operatorStateRetriever(),
+            _registryCoordinator(),
+            signedBatch,
+            blobInclusionInfo,
+            _securityThresholds(),
+            _quorumNumbersRequired()
+        );
+        CertLib.revertOnError(err, errParams);
+    }
+
+    /**
+     * @notice Thin try/catch wrapper around verifyDACertV2 that returns false instead of reverting
+     * @dev Useful for ZK proof systems that require a return value
+     * @param batchHeader The batch header of the blob
+     * @param blobInclusionInfo The inclusion proof for the blob cert
+     * @param nonSignerStakesAndSignature The nonSignerStakesAndSignature to verify the blob cert against
+     * @param signedQuorumNumbers The signed quorum numbers corresponding to the nonSignerStakesAndSignature
+     * @return success True if verification succeeded
+     */
+    function verifyDACertV2ForZKProof(
```

Comment:
don't like this ZKProof suffix (also doesnt match V1 which is ZkProof). See my comment in v1 contract